perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -659,12 +659,23 @@ order of title.
 
 sub response_templates {
     my $problem = shift;
-    return $problem->result_source->schema->resultset('ResponseTemplate')->search(
+    my $schema = $problem->result_source->schema;
+    my @cids = $schema->resultset("Contact")->search(
         {
-            body_id => $problem->bodies_str_ids
+            body_id => $problem->bodies_str_ids,
+            category => $problem->category
+        }, {
+            columns => [ 'id' ],
+        }
+    )->get_column('id')->all;
+    return $schema->resultset('ResponseTemplate')->search(
+        {
+            body_id => $problem->bodies_str_ids,
+            contact_id => [ \@cids, undef ],
```

Comment:
I'm not sure about including templates with no category set here. Is your intention that not ticking any category in the template admin form implicitly associate it with all categories? Would a 'select all' button in the template editor (and perhaps validation to make sure at least one category is selected) make this clearer?